### PR TITLE
Fix 14.1.4 -> 15.0.3 bump

### PR DESCRIPTION
**Pull request by davthoNordnet**
image

image

image

image

### CODE REVIEW

File: src/Molecules/Input/FormattedNumber/FormattedNumber.tsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+      onStepUp?.();
+    } else {
+      onStepDown?.();
+    }
+
+    inputRef?.current?.focus();
+  };
+
+  const onKeyDownHandler = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    const { key } = e;
```
Comment (davthoNordnet) [inline src/Molecules/Input/FormattedNumber/FormattedNumber.tsx:93]: not quite null checked, potential problem here

File: src/Molecules/Input/FormattedNumber/FormattedInput/utils.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  const replaceMinusSignRegExp = new RegExp(escapeStringRegexp(minusSign), 'g');
+
+  const numericValue = inputValue.replace(nonNumericRegExp, replaceValue);
+
+  const delocalizedValue = numericValue
+    .replace(replaceDecimalRegExp, '.') // replace localized decimalSign
+    .replace(replaceMinusSignRegExp, '-'); // replace localized minusSign
+
+  const sanitizedValue = delocalizedValue
+    .replace(/(?<=^|-)\./, '0.') // prepend 0 when integer part is missing for decimal
```
Comment (davthoNordnet) [inline src/Molecules/Input/FormattedNumber/FormattedInput/utils.ts:42]: I'm no regex-expert but I read online that this regex could cause issues for safari users. Here's the suggested solution

```suggestion
    .replace(/(?:^|-)\./, '0.') // prepend 0 when integer part is missing for decimal
```

File: src/Molecules/Input/FormattedNumber/FormattedInput/utils.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    .replace(/(?<!^)-/g, replaceValue); // remove minusSign not at start of string
```
Comment (davthoNordnet) [inline src/Molecules/Input/FormattedNumber/FormattedInput/utils.ts:43]: This is also a lookbehind-regex that is not supposed to be supported in older safari-versions

File: src/Molecules/Input/FormattedNumber/FormattedInput/utils.ts [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+  return indexOfLastDigit + 1;
+};
+
+export const handleThousandSeparatorRemoval = (
+  e: React.KeyboardEvent<HTMLInputElement>,
+  intl: IntlShape,
+) => {
+  const el = e.target as HTMLInputElement;
+  const { key } = e;
```
[the remark this answers is not in the document]
Comment (TahirMehdi) [reply src/Molecules/Input/FormattedNumber/FormattedInput/utils.ts:147]: Target should always be HTMLInputElement, we have tests covering that.

File: src/Molecules/Input/FormattedNumber/FormattedInput/utils.ts (hunk 2 above)
Comment (TahirMehdi) [reply src/Molecules/Input/FormattedNumber/FormattedInput/utils.ts:42]: Fixed

**Pull request closed without merging**